urlcdd0c1ab49_emiln_JHouseFireExplorer_tgz-pJ8-dk_brics_automaton_DatatypesJ8/src/dk/brics/automaton/BasicAutomata.java

```diff
@@ -0,0 +1,488 @@
+			b.append(c);
+			maxInteger(n, i + 1, b);
+		}
+		b.append(')');
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing nonnegative integers
+	 * that are not less that the given value.
+	 * @param n string representation of minimum value
+	 */
+	public static Automaton makeMinInteger(String n) {
+		int i = 0;
+		while (i + 1 < n.length() && n.charAt(i) == '0')
+			i++;
+		StringBuilder b = new StringBuilder();
+		b.append("0*");
+		minInteger(n.substring(i), 0, b);
+		b.append("[0-9]*");
+		return Automaton.minimize((new RegExp(b.toString())).toAutomaton());
+	}
+	
+	private static void minInteger(String n, int i, StringBuilder b) {
+		b.append('(');
+		if (i < n.length()) {
+			char c = n.charAt(i);
+			if (c != '9')
+				b.append("[" + (char)(c+1) + "-9][0-9]{" + (n.length() - i - 1) + "}|");
+			b.append(c);
+			minInteger(n, i + 1, b);
+		}
+		b.append(')');
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing decimal numbers
+	 * that can be written with at most the given number of digits.
+	 * Surrounding whitespace is permitted.
+	 * @param i max number of necessary digits
+	 */
+	public static Automaton makeTotalDigits(int i) {
+		return Automaton.minimize((new RegExp("[ \t\n\r]*[-+]?0*([0-9]{0," + i + "}|((([0-9]\\.*){0," + i + "})&@\\.@)0*)[ \t\n\r]*")).toAutomaton());
+	}
+	
+	/**
+	 * Constructs automaton that accept strings representing decimal numbers
+	 * that can be written with at most the given number of digits in the fraction part.
+	 * Surrounding whitespace is permitted.
+	 * @param i max number of necessary fraction digits
+	 */
+	public static Automaton makeFractionDigits(int i) {
+		return Automaton.minimize((new RegExp("[ \t\n\r]*[-+]?[0-9]+(\\.[0-9]{0," + i + "}0*)?[ \t\n\r]*")).toAutomaton());
+	}
+	
+	/**
+	 * Constructs automaton that accept strings representing the given integer.
+	 * Surrounding whitespace is permitted.
+	 * @param value string representation of integer
+	 */
+	public static Automaton makeIntegerValue(String value) {
+		boolean minus = false;
+    	int i = 0;
+    	while (i < value.length()) {
+    		char c = value.charAt(i);
+    		if (c == '-')
+    			minus = true;
+    		if (c >= '1' && c <= '9')
+    			break;
+    		i++;
+    	}
+    	StringBuilder b = new StringBuilder();
+		b.append(value.substring(i));
+		if (b.length() == 0)
+			b.append("0");
+		Automaton s;
+		if (minus)
+			s = Automaton.makeChar('-');
+		else
+			s = Automaton.makeChar('+').optional();
+		Automaton ws = Datatypes.getWhitespaceAutomaton();
+		return Automaton.minimize(ws.concatenate(s.concatenate(Automaton.makeChar('0').repeat()).concatenate(Automaton.makeString(b.toString()))).concatenate(ws));		
+	}
+	
+	/**
+	 * Constructs automaton that accept strings representing the given decimal number.
+	 * Surrounding whitespace is permitted.
+	 * @param value string representation of decimal number
+	 */
+	public static Automaton makeDecimalValue(String value) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.